src/pages/schedule/index.js

```diff
@@ -1,72 +1,108 @@
 import React, { Component } from "react";
+import moment from 'moment-timezone'
 import Layout from "../../components/layout";
 import Alternate from "../../components/Alternate";
-import scheduleData from "../../content/schedule.json";
+import rawScheduleData from "../../content/schedule.json";
 import Text from "../../components/Text";
 
-const convertTime = (time) => {
-    time = time.split(/[: ]/);
-    if (time[2] === "PM" && time[0] != 12) time[0] = parseInt(time[0]) + 12;
-    let d = new Date();
-    d.setHours(time[0]);
-    return d;
+const DEFAULT_TIMEZONE = 'EST'
+const TIMEZONE_STORAGE_KEY = 'schedule.timezone'
+
+const SCHEDULE_WITH_TIMES_PARSED = rawScheduleData.map((entry) => ({
+    ...entry,
+    time: moment(`2020-08-14 ${entry.USEDT} EDT`)
+}))
+
+const TIMEZONES = {
+    "US/Eastern":"Eastern",
+    "US/Central":"Central",
+    "US/Mountain":"Mountain",
+    "US/Pacific":"Pacific",
+    "US/Alaska":"Alaska",
+    "US/Hawaii":"Hawaii",
 }
 
+console.log(SCHEDULE_WITH_TIMES_PARSED[0].time.format())
+
 const timeSort = (a, b) => {
-    if (a.USEST === b.USEST) {
+    if (a.time.unix() === b.time.unix()) {
         return a.TrackLink > b.TrackLink ? 1 : -1;
     }
-    a.time = convertTime(a.USEST);
-    b.time = convertTime(b.USEST);
-    return a.time > b.time ? 1 : -1;
+    return a.time.unix() > b.time.unix() ? 1 : -1;
 }
 
 const trackSort = (a, b) => {
     if (a.TrackLink === b.TrackLink) {
-        a.time = convertTime(a.USEST);
-        b.time = convertTime(b.USEST);
-        return a.time > b.time ? 1 : -1;
+        return a.time.unix() > b.time.unix() ? 1 : -1;
     }
     return a.TrackLink > b.TrackLink ? 1 : -1;
 }
 
+const getInitialTimezone = () => {
+    const stored = window.localStorage?.getItem?.(TIMEZONE_STORAGE_KEY)
+    return stored || DEFAULT_TIMEZONE
+}
+
+const persistTimezone = (timezone) =>{
+    // eslint-disable-next-line
+    window.localStorage?.setItem?.(TIMEZONE_STORAGE_KEY, timezone)
```

Comment:
this is a very defensive function call, and eslint gets confused with optional chaining function calls. I could have used the long syntax, `window.localStorage && window.localStorage.setItem && ... .setItem()`, but I wanted to let webpack do that for me with optional chaining.